codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -880,12 +876,17 @@ public int nextStreamId() {
         }
 
         @Override
-        public boolean createdStreamId(int streamId) {
+        public boolean isStreamForEndpoint(int streamId) {
             boolean even = (streamId & 1) == 0;
             return server == even;
         }
 
         @Override
+        public boolean mayHaveCreatedStream(int streamId) {
+            return isStreamForEndpoint(streamId) && streamId <= lastStreamCreated;
```

Comment:
Ok just to clarify...we will only be doing it in 1 spot?  Or define another `private static` maybe?